doc/classes/Object.xml

```diff
@@ -170,23 +217,23 @@
 			<param index="1" name="callable" type="Callable" />
 			<param index="2" name="flags" type="int" default="0" />
 			<description>
-				Connects a [param signal] to a [param callable]. Use [param flags] to set deferred or one-shot connections. See [enum ConnectFlags] constants.
-				A signal can only be connected once to a [Callable]. It will print an error if already connected, unless the signal was connected with [constant CONNECT_REFERENCE_COUNTED]. To avoid this, first, use [method is_connected] to check for existing connections.
-				If the callable's target is destroyed in the game's lifecycle, the connection will be lost.
+				Connects a [param signal] by name to a [param callable]. Optional [param flags] can be also added to configure the connection's behavior (see [enum ConnectFlags] constants).
+				A signal can only be connected once to the same [Callable]. If the signal is already connected, returns [constant ERR_INVALID_PARAMETER] and pushes an error message, unless the signal is connected with [constant CONNECT_REFERENCE_COUNTED]. To prevent this, use [method is_connected] first to check for existing connections.
+				If the [param callable]'s object is freed, the connection will be lost.
 				[b]Examples with recommended syntax:[/b]
-				Connecting signals is one of the most common operations in Godot and the API gives many options to do so, which are described further down. The code block below shows the recommended approach for both GDScript and C#.
+				Connecting signals is one of the most common operations in Godot and the API gives many options to do so, which are described further down. The code block below shows the recommended approach.
 				[codeblocks]
 				[gdscript]
 				func _ready():
 				    var button = Button.new()
-				    # `button_down` here is a Signal object, and we thus call the Signal.connect() method,
-				    # not Object.connect(). See discussion below for a more in-depth overview of the API.
+				    # `button_down` here is a Signal variant type, and we thus call the Signal.connect() method, not Object.connect().
+				    # See discussion below for a more in-depth overview of the API.
 				    button.button_down.connect(_on_button_down)
 
-				    # This assumes that a `Player` class exists which defines a `hit` signal.
+				    # This assumes that a `Player` class which defines a `hit` signal exists.
```

Comment:
The same sentence exists below on line 252, could use the same edit for consistency!

Reply:
Same on 316!

Reply:
I kind of reverted the wording a bit here and added a comma. In retrospective it didn't flow right.